keras_cv/layers/object_detection/anchor_generator.py

```diff
@@ -0,0 +1,272 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+from tensorflow import keras
+
+from keras_cv import bounding_box
+
+
+class AnchorGenerator(keras.layers.Layer):
+    """AnchorGenerator generates anchors for multiple feature maps.
+
+    AnchorGenerator takes multiple scales and generates anchor boxes based on the anchor
+    sizes, scales, aspect ratios, and strides provided.  To invoke AnchorGenerator, call
+    it on the image that needs anchor boxes.
+
+    `anchor_sizes` and `strides` must match structurally - they are pairs.  Scales and a
+    spect ratios can either be a list, that is then used for all of the sizes
+    (aka levels), or a dictionary from `{'level_{number}': [parameters at scale...]}`.
+
+    Args:
+      bounding_box_format: The format of bounding boxes to generate. Refer
```

Comment:
Should we have a default value for this? Most of the time it should be xyxy and users need to convert to xywh before regression targets

Reply:
We don't have any default values for bounding_box_format in our API at the moment.  Once we add them - it is a one-way change.

We decided awhile ago it just opens up the potential for too many silent errors.  Additionally, while it makes code easier to write, it makes it harder to read;


```
anchor_generator = AnchorGenerator()
retina_net = RetinaNet(anchor_generator=anchor_generator)
```

vs

```
anchor_generator = AnchorGenerator(bounding_box_format='xywh')
retina_net = RetinaNet(anchor_generator=anchor_generator)
```

Is the format correct for the retina net in sample 1?  Its not clear to code-readers - even if the writer looked at the docstring and made sure they match.  This makes debugging a huge pain.